runtime/lua/vim/net.lua

```diff
@@ -0,0 +1,53 @@
+local M = {}
+
+--- Downloads a file from the given URL to a specified location.
+---
+--- @param url string The URL to download from.
+--- @param download_location string The file path where the content should be saved.
+--- @param opts? table Optional parameters:
+---   - on_exit (function|nil): Called when the download completes or fails. Receives error string or nil.
+---   - verbose (boolean|nil): Enables verbose curl output.
+---   - retry (integer|nil): Number of times to retry on transient failures (default: 3).
+---
+--- @return nil
+function M.download(url, download_location, opts)
```

Comment:
Please make the callback the last positional argument.

```suggestion
--- Downloads a file from the given URL to a specified location.
---
--- @param url string The URL to download from.
--- @param output string The file path where the content should be saved.
--- @param opts? table Optional parameters:
---   - verbose (boolean|nil): Enables verbose curl output.
---   - retry (integer|nil): Number of times to retry on transient failures (default: 3).
--- @param on_exit fun(err?: string)
--- @return nil
function M.download(url, output, opts, on_exit)
```

We may also want this function to operate synchronously if no callback is given.

Also I think we may want a variation of this function where the output is passed back as a string. For this case `on_exit` should have type `fun(data?: string, err?: string)`

Reply:
I will add the synchronous behaviour, thank you.

> Also I think we may want a variation of this function where the output is passed back as a string. For this case on_exit should have type fun(data?: string, err?: string)

Yes, I believe we should implement this in a future PR as a separate function, something like `vim.net.request()`.